static/src/javascripts/projects/common/modules/commercial/first-pv-consent-plus-fiv-banner.js

```diff
@@ -0,0 +1,136 @@
+                    </div>
+                </div>
+            </div>
+        </div>
+    
+        <div class="js-first-pv-consent-site-message site-message--first-pv-consent" tabindex="-1" data-link-name="release message" role="dialog" aria-label="welcome" aria-describedby="site-message__message">
+            <div class="gs-container">
+                <div class="site-message__inner js-site-message-inner">
+                    <div class="site-message__roundel">
+                        ${marque36icon.markup}
+                    </div>
+                    <div class="site-message__copy js-site-message-copy u-cf">
+                        ${makeFirstPvConsentHtml()}
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+`;
+
+class SubMessage extends Message {
+    elementSelector: string;
+    parent: Message;
+
+    constructor(
+        id: string,
+        elementSelector: string,
+        parent: Message,
+        options?: Object
+    ) {
+        super(id, options);
+        this.elementSelector = elementSelector;
+        this.parent = parent;
+    }
+
+    hide(): void {
+        const element = document.querySelector(this.elementSelector);
+        if (element) {
+            const parentElement = element.parentElement;
+            element.remove();
+            if (parentElement && parentElement.childElementCount === 0) {
+                parentElement.remove();
+            }
+        }
+
+        // Don't display parent again if any sub-message has been hidden
+        this.parent.remember();
+    }
+
+    bindCloseHandler(): void {
+        const element = document.querySelector(this.elementSelector);
+        if (element) {
+            const closeButton = element.querySelector('.js-site-message-close');
+            if (closeButton) {
+                closeButton.addEventListener('click', () => {
+                    this.acknowledge();
+                });
+            }
+        }
+    }
+}
+
+const firstPvFivPlusfivMessage = new Message(messageCode);
+
+const firstPvConsentMessage = new SubMessage(
+    firstPvConsentMessageCode,
+    '.js-first-pv-consent-site-message',
+    firstPvFivPlusfivMessage
+);
+const fivMessage = new SubMessage(
+    fivMessageCode,
+    '.js-fiv-banner-site-message',
+    firstPvFivPlusfivMessage
+);
+
+const show = (): Promise<boolean> => {
+    const fivBannerHtml = getFivBannerHtml(bannerParams);
+    const bannerHtml = getBannerHtml(fivBannerHtml);
+
+    trackFirstPvConsent();
+    if (document.body) {
+        document.body.insertAdjacentHTML('beforeend', bannerHtml);
+    }
+    bindFirstPvConsentClickHandlers(firstPvConsentMessage);
+    fivMessage.bindCloseHandler();
+
+    return Promise.resolve(true);
+};
+
+const firstPvConsentPlusFivBanner: Banner = {
+    id: messageCode,
+    canShow: (): Promise<boolean> =>
+        Promise.all([canShowFirstPvConsent(), canShowFivBanner()]).then(
```

Comment:
I assume the switch is part of the `canShow` for the fiv banner? just checking this won't show if the switch is off!

Reply:
yes, these are imported at the top from the `canShow` methods on `fiv-banner` and `first-pv-consent-banner`